x/staking/types/delegation.go

```diff
@@ -315,14 +316,21 @@ func (d Redelegation) String() string {
   Delegator:                 %s
   Source Validator:          %s
   Destination Validator:     %s
-  Entries:`, d.DelegatorAddress, d.ValidatorSrcAddress, d.ValidatorDstAddress)
+  Entries:
+`,
```

Comment:
When printing this out, the output may carry an extra `\n`. I'd revert this

Reply:
No, this is intentional. The output is not structured correctly otherwise.

Reply:
Still: `fmt.Printf("%s\n", obj)` would return double new line terminated strings, which is quite inconsistent with some parts of the codebase, see `x/gov/proposal.go`:

```golang
func (p Proposal) String() string {
	return fmt.Sprintf(`Proposal %d:
  Title:              %s
  Type:               %s
  Status:             %s
  Submit Time:        %s
  Deposit End Time:   %s
  Total Deposit:      %s
  Voting Start Time:  %s
  Voting End Time:    %s
  Description:        %s`,
		p.ProposalID, p.GetTitle(), p.ProposalType(),
		p.Status, p.SubmitTime, p.DepositEndTime,
		p.TotalDeposit, p.VotingStartTime, p.VotingEndTime, p.GetDescription(),
	)
}
```

Adding terminating new lines shouldn't be on the type's `String()` function anyway. It should be on the function responsibile to print to file.

Reply:
> ... would return double new line terminated strings ...
@alessio you're misunderstanding. No it wouldn't. There is additional code after this that adds to the output:

```go
	for i, entry := range r.Entries { // ... }
```

Reply:
Here is the exact output:

```
Redelegations between:
  Delegator:                 cosmos1m9547m99vs8e9f40acs7tveg7zuxzrhhcpezc6
  Source Validator:          cosmosvaloper1m9547m99vs8e9f40acs7tveg7zuxzrhha4dh5f
  Destination Validator:     cosmosvaloper1hsktdq8ftxj4dx5ndnpcee9rg6evd95xtmh5kd
  Entries:
    Redelegation Entry #0:
      Creation height:           409
      Min time to unbond (unix): 2019-05-17 19:42:14.5681452 +0000 UTC
      Initial Balance:           35
      Shares:                    35.000000000000000000
      Balance:                   35
    Redelegation Entry #1:
      Creation height:           471
      Min time to unbond (unix): 2019-05-17 19:47:54.4826337 +0000 UTC
      Initial Balance:           35
      Shares:                    35.000000000000000000
      Balance:                   35
    Redelegation Entry #2:
      Creation height:           490
      Min time to unbond (unix): 2019-05-17 19:49:39.1091959 +0000 UTC
      Initial Balance:           22
      Shares:                    22.000000000000000000
      Balance:                   22
```

Please tell me what is wrong?

Reply:
True, but...

Reply:
The last line is empty as in the string contains two trailing `\n`.

Reply:
Because of https://github.com/cosmos/cosmos-sdk/pull/4333#discussion_r284005193

Reply:
I trimmed the last newline (after the last redelegation entry).